### PR TITLE
Build to a single file

### PR DESCRIPTION
**Pull request by matthewp**
## Changes

- Changes the build so that there is 1 single bundle for the entire app.
- This is needed for some SSR hosts.
- Also might speed up static builds.
- Fixes endpoint routes in SSR.
- The change is to remove the use of `facadeId` maps in the internals module. Instead we put metadata about a route in the existing `BuildPageData` object.
- `vite-plugin-pages` is the entry module for the build. It imports all routes and prepares itself for static rendering / ssr.
- `vite-plugin-ssr` is the entry for SSR. It imports the vite-plugin-pages module and implements the adapter API.

## Testing

- Test added for SSR endpoint routes.
- Otherwise this is mostly just a refactor.

## Docs

Not yet, separate issue.

### CODE REVIEW

File: packages/astro/src/core/app/index.ts [hunk 1]
```diff
@@ -1,31 +1,34 @@
...
 import { defaultLogOptions } from '../logger.js';
 export { deserializeManifest } from './common.js';
 import { matchRoute } from '../routing/match.js';
 import { render } from '../render/core.js';
+import { call as callEndpoint } from '../endpoint/index.js';
 import { RouteCache } from '../render/route-cache.js';
 import { createLinkStylesheetElementSet, createModuleScriptElementWithSrcSet } from '../render/ssr-element.js';
 import { prependForwardSlash } from '../path.js';
 
+const supportedFileNameToMimeTypes = new Map<string, string>([
```
Comment (matthewp) [inline packages/astro/src/core/app/index.ts:13]: Try maybe using a mime type package instead.

File: packages/astro/src/core/app/index.ts [hunk 2]
```diff
@@ -1,31 +1,34 @@
...
+	constructor(manifest: Manifest) {
 		this.#manifest = manifest;
 		this.#manifestData = {
 			routes: manifest.routes.map((route) => route.routeData),
 		};
-		this.#rootFolder = rootFolder;
 		this.#routeDataToRouteInfo = new Map(manifest.routes.map((route) => [route.routeData, route]));
 		this.#routeCache = new RouteCache(defaultLogOptions);
-		this.#renderersPromise = this.#loadRenderers();
+		this
```
Comment (matthewp) [inline packages/astro/src/core/app/index.ts:31]: remove

File: packages/astro/src/core/app/index.ts [hunk 3]
```diff
@@ -80,26 +94,47 @@ export class App {
...
+			pathname: url.pathname,
+			routeCache: this.#routeCache,
+			ssr: true,
+		});
+
+		if(result.type === 'response') {
+			return result.response;
+		} else {
+			const body = result.body;
+			const ext = /\.([a-z]+)/.exec(url.pathname);
```
Comment (matthewp) [inline packages/astro/src/core/app/index.ts:124]: try to do this only once.

**Comment by natemoo-re** [comment]
Can't seem to leave a review right now, but looks good! 

Saw one stray `console.log` at https://github.com/withastro/astro/blob/e92e661990f9aa1c2be04867e6aeb47deed63418/packages/astro/src/core/build/static-build.ts#L105

**Pull request merged**